### DISCUSSION TITLE
Does Deduplication include also Delta Encoding?

### DISCUSSION THREAD

**Discussion by Tom-H-L** [borgbackup/borg#7113] [Q&A] [answered]
Hi all,

Does deduplication only mean that multiple identical files are backuped only once or does it also include delta encoding, i.e. that only those parts within one file are backuped that are different to the previous backup? With other words: does borg support deduplication AND delta backups AND compression to have the smallest possible incremental/ differential backups possible?

**Comment by infectormp** [comment, marked as answer]
> https://borgbackup.readthedocs.io/en/stable/internals.html

Deduplication is performed globally across all data in the repository (multiple backups and even multiple hosts), both on data and file metadata, using [Chunks](https://borgbackup.readthedocs.io/en/stable/internals/data-structures.html#chunks) created by the chunker using the [Buzhash](https://en.wikipedia.org/wiki/Buzhash) algorithm (“buzhash” chunker) or a simpler fixed blocksize algorithm (“fixed” chunker).
